src/routers/user/index.ts

```diff
@@ -192,4 +197,37 @@ router.delete('/:id', auth, async (req: Request, res: Response) => {
   }
 });
 
+// check membership by email or watiam user id
+router.post(
+  '/membership-check',
+  routeValidator('/membership-check'),
+  validate,
+  async (req: Request, res: Response, next: NextFunction) => {
+    try {
+      const { emailOrWatIAMUserId } = req.body as MembershipCheckRequestBody;
+
+      const isEmail = validator.isEmail(emailOrWatIAMUserId);
+      const conditions: Record<string, string> = {};
+      if (isEmail) {
+        conditions.email = emailOrWatIAMUserId;
+      } else {
+        conditions.watIAMUserId = emailOrWatIAMUserId;
+      }
+      const user = await UserModel.findOne(conditions);
+
+      if (user === null) {
+        throw createError(
+          404,
+          `${
+            isEmail ? 'Email' : 'WatIAM user id'
```

Comment:
Can use `validator.isEmail(emailOrWatIAMUserId);` here I think. I don't think this route has to care too much about which of the 2 they sent...

What do u think?

Reply:
Might be misunderstanding. `isEmail` is `validator.isEmail(emailOrWatIAMUserId)` I just put it in a variable since I used it twice.

We could just send back "Email or WatIAM user id provided does not match an existing user".  I don't think it hurts to specify tho and it might make it a lil nicer if we want to display the error in the client.

Reply:
Yea I like that check 4 sure I think we should keep em.

This was related to the comment above. What I was trying to say is that we should keep this check, but instead of using `isEmail`, we can use `validator.isEmail(emailOrWatIAMUserId)` because the check to distinguish between email and UserId is more valuable here than when you where calling `findOne()`

But I explained poorly🤣

Reply:
I'm rtrdd